themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,273 @@
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="ex-expand1"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell key={'app-' + appIndex}>
+                        <span id="check-action-item1">
+                          <BuilderImageIcon/> {application.clientId}
+                          </span>
+                        </DataListCell>,
+                        <DataListCell key={'internal-' + appIndex}>
+                          <span>{application.internal ? 'Internal' : 'Third-party'}</span>
+                        </DataListCell>,
+                        <DataListCell key={'status-' + appIndex}>
+                          <span>{application.inUse ? 'In-use' : 'NaN'}</span>
+                        </DataListCell>,
+                        <DataListCell key={'url-' + appIndex}>
+                          <a href={application.url}> <LinkIcon/> {application.url}</a>
+                        </DataListCell>
+                    ]}
+                  />
+                </DataListItemRow>
+                <DataListContent 
+                  noPadding={false}
+                  aria-label="Application Details"
+                  id="ex-expand1"
+                  isHidden={!this.state.isRowOpen[appIndex]}
+                >
+                  <div className="pf-c-content">
+                    <h3>Client</h3>
+                    <hr/>
+                    <h3>Description</h3>
+                    <h3>URL</h3>
+                    <h3>Has access to</h3>
+                    <h3>Access granted on</h3>
+                    <hr/>
+                    <React.Fragment>
+                      <ContinueCancelModal 
+                        buttonTitle='Remove' // required
+                        buttonVariant='secondary' // defaults to 'primary'
+                        modalTitle='Remove Access' // required
+                        modalMessage='This will remove the currently granted access permission for AdminCLI. You will need to grant access again if you want ot user this app.'
+                        modalContinueButtonLabel='Confirm' // defaults to 'Continue'
+                        modalCancelButtonLabel='Cancel' // defaults to 'Cancel'
+                        onContinue={this.doSomething} // required
+                      />
+                    </React.Fragment>
+                    <InfoAltIcon/> 
+                    <span>
+                      By clicking ‘Remove Access’, you will remove granted permissions of this app. This app will no longer use your information.
+                    </span>
+                  </div>
+                </DataListContent>
+              </DataListItem>
+            )
+          })}
+        </DataList>
+      </ContentPage>
+    );
+  }
+};
+
+interface DropdownWrapperProps {
+  dropdownWrapperItems: any[];
```

Comment:
Never use 'any'

Reply:
@ssilvert I got the inspiration from https://github.com/patternfly/patternfly-react/blob/9335f73f55ac774b4d65eb353e149cf973164293/packages/patternfly-4/react-core/src/components/Dropdown/Dropdown.tsx#L12. Should we really change it? Assuming that the component does the same?

Reply:
We should change it.  'any' defeats the whole purpose of TypeScript.  Furthermore, it will fail our linting rules.

Don't worry.  Once you get the hang of it, you will never want to use 'any' again.